signer/src/stacks/api.rs

```diff
@@ -634,76 +626,116 @@ impl StacksClient {
+                nakamoto_start_height = %stacks.nakamoto_start_height(),
+                last_chain_length = %block.header.chain_length,
+                "Stopping, since we have fetched all Nakamoto blocks"
+            );
+            break;
         }
-        Err(Error::StacksNodeResponse(e)) => {
-            if e.is_timeout() || e.is_connect() {
-                client.next_endpoint();
-            }
+        // We've seen this parent already, so time to stop.
+        if db.stacks_block_exists(block.header.parent_block_id).await? {
+            tracing::info!("Parent block known in the database");
+            break;
         }
-        _ => (),
+        // There are more blocks to fetch, so let's get them.
+        let mut tenure_blocks = stacks.get_tenure(block.header.parent_block_id).await?;
+        blocks.append(&mut tenure_blocks);
     }
-    result
+
+    Ok(blocks)
+}
+
+/// A deserializer for Clarity's [`Value`] type that deserializes a hex-encoded
+/// string which was serialized using Clarity's consensus serialization format.
+fn clarity_value_deserializer<'de, D>(deserializer: D) -> Result<Value, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    Value::try_deserialize_hex_untyped(&String::deserialize(deserializer)?)
+        .map_err(serde::de::Error::custom)
 }
 
 impl StacksInteract for StacksClient {
     async fn get_current_signer_set(
-        &mut self,
+        &self,
         contract_principal: &StacksAddress,
     ) -> Result<Vec<PublicKey>, Error> {
         // Make a request to the sbtc-registry contract to get the current
         // signer set.
-        let result = StacksClient::get_data_var(
-            self,
-            contract_principal,
-            &ContractName::from("sbtc-registry"),
-            &ClarityName::from("current-signer-set"),
-        )
-        .await;
-
-        let checked_result = check_result(self, result)?;
+        let result = self
+            .get_data_var(
+                contract_principal,
+                &ContractName::from("sbtc-registry"),
+                &ClarityName::from("current-signer-set"),
+            )
+            .await?;
 
-        match checked_result {
+        // Check the result and return the signer set. We're expecting a
+        // list of buffers, where each buffer is a public key.
+        match result {
             Value::Sequence(SequenceData::List(ListData { data, .. })) => {
-                let mut keys = Vec::new();
-                for item in data {
-                    if let Value::Sequence(SequenceData::Buffer(BuffData { data })) = item {
-                        keys.push(PublicKey::from_slice(&data)?);
-                    } else {
-                        return Err(Error::InvalidStacksResponse(
+                // Iterate through each record in the list and verify that it's a buffer.
+                // If it is a buffer, then convert it to a public key.
+                // Otherwise, abort retries and return an error.
```

Comment:
Should be "Otherwise, return an error".

Reply:
Addressed in [5f48dd2](https://github.com/stacks-network/sbtc/pull/572/commits/5f48dd257ab134735de76327fa520789f57ed27d)